test-runner-poc/lib/runner.js

```diff
@@ -1,73 +1,140 @@
-"use strict";
+'use strict';
 
-const path = require("path");
-const fs = require("fs");
-const utils = require("mocha").utils;
+const path = require('path');
+const fs = require('fs');
+const MongoClient = require('mongodb').MongoClient;
+const parseConnectionString = require('../../lib/core/uri_parser');
 
+const testPath = path.join(path.join(process.cwd(), '../'), 'test');
+const configPath = path.join(testPath, 'config.js');
+const envPath = path.join(testPath, 'environments.js');
+const environments = require(envPath);
+const TestConfiguration = require(configPath);
+
+const mock = require('mongodb-mock-server');
+
+let mongoClient;
 let filters = [];
-let files = [];
 
 function addFilter(filter) {
-	if (typeof filter !== "function" && typeof filter !== "object") {
-		throw new Error(
-			"Type of filter must either be a function or an object"
-		);
-	}
-	if (
-		typeof filter === "object" &&
-		(!filter.filter || typeof filter.filter !== "function")
-	) {
-		throw new Error("Object filters must have a function named filter");
-	}
-
-	if (typeof filter === "function") {
-		filters.push({ filter: filter });
-	} else {
-		filters.push(filter);
-	}
+  switch (typeof filter) {
+    case 'function':
+      filters.push({ filter: filter });
+      break;
+    case 'object':
+      if (!filter.filter || typeof filter.filter !== 'function') {
+        throw new Error('Object filters must have a function named filter');
+      }
+      filters.push(filter);
+      break;
+    default:
+      throw new Error('Type of filter must either be a function or an object');
+  }
 }
 
-before(function() {
+function environmentSetup(environmentCallback) {
+  const mongodb_uri = process.env.MONGODB_URI || 'mongodb://localhost:27017';
+  mongoClient = new MongoClient(mongodb_uri);
+
+  mongoClient.connect((err, client) => {
+    if (err) throw new Error(err);
+
+    createFilters(callback);
+
+    function callback(environmentName, version) {
```

Comment:
Can we use a more descriptive function name?

Reply:
renamed to `environmentParser`